crates/sparrow-main/tests/e2e/collection_tests.rs

```diff
@@ -0,0 +1,124 @@
+            {"time": 3000, "key": 1, "e0": {"f2": 13},           "e1": 42, "e2": null, "e3": "f2" }
+            {"time": 4000, "key": 1, "e0": {"f1": 15, "f3": 11}, "e1": 3,  "e2": 7,    "e3": "f3" }
+            "#;
+    input
+}
+
+#[tokio::test]
+// #[ignore]
+async fn test_get_static_key() {
+    insta::assert_snapshot!(QueryFixture::new("{ f1: get(Input.e0, \"f1\") }").run_to_csv(&arrow_collection_data_fixture().await).await.unwrap(), @r###"
+    _time,_subsort,_key_hash,_key,m,n,eq
+    "###);
+}
+
+#[tokio::test]
+#[ignore]
+async fn test_get_data_key() {
+    insta::assert_snapshot!(QueryFixture::new("{ value: Input.map | get(Input.key)} }").run_to_csv(&arrow_collection_data_fixture().await).await.unwrap(), @r###"
+    _time,_subsort,_key_hash,_key,m,n,eq
+    "###);
+}
+
+fn batch_from_json(json: &str, column_types: Vec<DataType>) -> anyhow::Result<RecordBatch> {
+    // Trim trailing/leading whitespace on each line.
+    let json = json.lines().map(|line| line.trim()).join("\n");
+
+    // Determine the schema.
+    let schema = {
+        let mut fields = vec![
+            Field::new(
+                "time",
+                DataType::Timestamp(TimeUnit::Nanosecond, None),
+                false,
+            ),
+            Field::new("key", DataType::UInt64, false),
+        ];
+
+        fields.extend(
+            column_types
+                .into_iter()
+                .enumerate()
+                .map(|(index, data_type)| Field::new(format!("e{index}"), data_type, true)),
+        );
+
+        Arc::new(Schema::new(fields))
+    };
+
+    // Create the reader
+    let reader = std::io::Cursor::new(json.as_bytes());
+    let reader = arrow::json::ReaderBuilder::new(schema).build(reader)?;
+
+    // Read all the batches and concatenate them.
+    let batches: Vec<_> = reader.try_collect()?;
+    let read_schema = batches.get(0).context("no batches read")?.schema();
+    let batch = arrow::compute::concat_batches(&read_schema, &batches)
+        .context("concatenate read batches")?;
+
+    Ok(batch)
+}
+
+async fn json_to_parquet_file(json_input: &str, file: File) {
+    let fields = Fields::from(vec![
+        Field::new("key", DataType::Utf8, false),
+        Field::new("value", DataType::Int64, false),
+    ]);
+
+    // TODO: need to figure out type equality - if this is changed to "map" field name it fails
+    let m1 = Arc::new(Field::new("map", DataType::Struct(fields), false));
+    let column_types = vec![
+        DataType::Map(m1, false),
+        DataType::Int64,
+        DataType::Float64,
+        DataType::Utf8,
+    ];
+    let record_batch = batch_from_json(json_input, column_types).unwrap();
+
+    // Create a Parquet writer
+    let mut writer = ArrowWriter::try_new(file, record_batch.schema(), None).unwrap();
+    writer.write(&record_batch).unwrap();
+
+    // Close the writer to finish writing the file
+    writer.close().unwrap();
+}
+
+async fn arrow_collection_data_fixture() -> DataFixture {
```

Comment:
I could move this to the `fixtures` class. It is currently creating a file each time, so I could technically just remove it, but it's been nice to be able to tweak the input data as I uncover more use cases I want to tests. So I'd prefer to leave it here for this PR, then remove in the next (or just move to `fixtures`) once I work on the implementation.

Reply:
https://github.com/kaskada-ai/kaskada/issues/493